src/Support/ClassLoader.php

```diff
@@ -135,28 +114,35 @@ class_alias($class, $reverse);
             return true;
         }
 
-        list($lowerClass, $upperClass, $lowerClassStudlyFile, $upperClassStudlyFile) = static::getPathsForClass($class);
-
-        foreach ($this->directories as $directory) {
-            $paths = [
-                $directory . DIRECTORY_SEPARATOR . $lowerClass,
-                $directory . DIRECTORY_SEPARATOR . $upperClass,
-                $directory . DIRECTORY_SEPARATOR . $lowerClassStudlyFile,
-                $directory . DIRECTORY_SEPARATOR . $upperClassStudlyFile,
-            ];
-
-            foreach ($paths as $path) {
-                if ($this->isRealFilePath($path)) {
-                    $this->includeClass($class, $path);
-
-                    if (!is_null($reverse = $this->getReverseAlias($class))) {
-                        if (!class_exists($reverse, false) && !in_array($reverse, $this->loadedAliases)) {
-                            class_alias($class, $reverse);
-                            $this->reversedClasses[] = $reverse;
+        // Check our registered autoload packages for a match
+        foreach ($this->autoloadedPackages as $prefix => $path) {
```

Comment:
The autoloaded packages should probably be sorted by the number of `\` characters in the prefix to make sure more specific prefixes are looked at first.